src/components/AgmExtension/ExtensionRequest.vue

```diff
@@ -157,6 +220,97 @@ export default class ExtensionRequest extends Vue {
     const data = { ...this.data, agmYear: value }
     this.emitData(data)
   }
+
+  get extensionExpiryMax (): string {
+    if (this.isFirstAgm) {
+      return DateUtilities.addMonthsToDate(30, DateUtilities.dateToYyyyMmDd(this.data.incorporationDate))
+    } else {
+      return 'placeholder for future work'
+    }
+  }
+
+  get extensionExpiryMin (): string {
+    if (this.isFirstAgm) {
+      return DateUtilities.addMonthsToDate(19, DateUtilities.dateToYyyyMmDd(this.data.incorporationDate))
+    } else {
+      return 'placeholder for future work'
+    }
+  }
+
+  /**
+   * Whether to disable the editing of AGM Year field.
+   * Editable (false) when Is this first AGM is set to No.
+   * Non-Editable (true) when we first load (null) or Is first AGM is Yes.
+   * Set the agmYear as incorporation date if not null and true.
+   */
+  get isFirstAgm (): boolean {
+    if (this.data.isFirstAgm !== null) {
+      if (this.data.isFirstAgm) {
+        this.data.agmYear = this.data.incorporationDate.getFullYear().toString()
+        return true
+      } else {
+        this.data.agmYear = ''
+        return false
+      }
+    } else {
+      return true
+    }
+  }
+
+  /** Called when extension date picker changes. */
+  @Watch('extensionExpiryDateText')
+  onExtensionDatePickerChanged (val: string): void {
+    this.data.prevExpiryDate = val
+  }
+
+  @Watch('intendedAgmDateText')
+  /** Called when intended AGM date picker changes. */
+  onAgmIntendedDatePickerChanged (val: string): void {
+    this.data.intendedAgmDate = val
+  }
+
+  @Watch('previousAgmDateText')
+  /** Called when previous AGM date picker changes. */
+  onPreviousAgmDatePickerChanged (val: string): void {
+    this.data.prevAgmDate = val
+  }
+
+  @Watch('data.isPrevExtension')
+  @Watch('extensionExpiryDateText')
+  onIsPrevExtensionChanged (val: boolean): void {
```

Comment:
val will be a boolean if `data.isPrevExtension` caused this change event.

val will be a string if `extensionExpiryDateText` caused this change event.

I recommend you don't use val and instead use the variable itself (eg, `this.data.isPrevExtension`).